### PR TITLE
CMake and code improvements/fixes

### PR DESCRIPTION
**Pull request by rpavlik**
Fixes the build on Linux, as much as I could right now (I didn't get the sphinx stuff building, but I should not have broken it), and make the CMake look modern, idiomatic, and clean.

Yes, this will probably require revisions to the sphinx in places, but it's much more idiomatic and simpler. The other stuff had a lot of feeling of "copy-pasted because this is what has worked in the past". Since people will copy and paste this as well, better if this is at least modern feeling and concise/non-duplicative.

Please consider reviewing and merging this. Rebasing it was a huge pain, and I really don't want to have to answer the github issues we'll get any time anyting in the OpenXR build changes that might affect the old build system.

### CODE REVIEW

**Comment by rpavlik** [comment]
OK, I rebased this and did a lot more clean up. It does actually build on linux and windows now on machines that don't have whatever special stuff the vendor has.

I also started on cleaning up how the shaders are compiled for chapter 5 but ran out of energy/time to copy it back to earlier chapters that need it. I did not update the sphinx docs that referred to the shader compilation/copying because they probably should have never even mentioned the details there to begin with, aside from "we added code to compile the shaders and copy them to the same directory as the binary". (Which, now that I think of it, has this been tested with the visual studio generator? because it's a multi-config generator so the .exe file is not in CMAKE_CURRENT_BINARY_DIR but a subdir)

File: .cmake-format.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# SPDX-FileCopyrightText: 2021-2023, Collabora, Ltd.
```
Comment (rbessems) [inline .cmake-format.py:1]: this should probably be Khronos MIT copyright.

Comment (rpavlik) [reply .cmake-format.py:1]: This is a copy and pasted config file. But that's why it's cc0, there's nothing meaningfully copyrightable in it, so yeah that can be changed.

Comment (rvkennedy) [reply .cmake-format.py:1]: @rbessems what would be the precise wording?

Comment (rbessems) [reply .cmake-format.py:1]: from issue 57:

```cpp
// Copyright 2023, The Khronos Group Inc.
//
// SPDX-License-Identifier: MIT
```

**Pull request merged**